app/src/main/java/com/nextcloud/talk/extensions/ImageViewExtensions.kt

```diff
@@ -0,0 +1,283 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+@file:Suppress("TooManyFunctions")
+
+package com.nextcloud.talk.extensions
+
+import android.graphics.drawable.Drawable
+import android.graphics.drawable.LayerDrawable
+import android.os.Build
+import android.util.Log
+import android.widget.ImageView
+import androidx.core.content.ContextCompat
+import androidx.core.content.res.ResourcesCompat
+import coil.annotation.ExperimentalCoilApi
+import coil.imageLoader
+import coil.load
+import coil.request.ImageRequest
+import coil.request.SuccessResult
+import coil.result
+import coil.transform.CircleCropTransformation
+import coil.transform.RoundedCornersTransformation
+import com.amulyakhare.textdrawable.TextDrawable
+import com.nextcloud.talk.R
+import com.nextcloud.talk.data.user.model.User
+import com.nextcloud.talk.ui.theme.ViewThemeUtils
+import com.nextcloud.talk.utils.ApiUtils
+
+private const val ROUNDING_PIXEL = 16f
+private const val TAG = "ImageViewExtensions"
+
+fun ImageView.loadAvatar(
+    user: User,
+    avatar: String,
+    requestBigSize: Boolean = true
+): io.reactivex.disposables
+.Disposable {
+
+    val imageRequestUri = ApiUtils.getUrlForAvatar(
+        user.baseUrl,
+        avatar,
+        requestBigSize
+    )
+
+    return loadAvatarInternal(user, imageRequestUri, false)
+}
+
+fun ImageView.replaceAvatar(
+    user: User,
+    avatar: String,
+    requestBigSize: Boolean = true
+): io.reactivex.disposables
+.Disposable {
+
+    val imageRequestUri = ApiUtils.getUrlForAvatar(
+        user.baseUrl,
+        avatar,
+        requestBigSize
+    )
+
+    return loadAvatarInternal(user, imageRequestUri, true)
+}
+
+@OptIn(ExperimentalCoilApi::class)
+private fun ImageView.loadAvatarInternal(
+    user: User?,
+    url: String,
+    replace: Boolean
+): io.reactivex.disposables
+.Disposable {
+
+    if (replace && this.result is SuccessResult) {
+        val result = this.result as SuccessResult
+        val memoryCacheKey = result.memoryCacheKey
+        val memoryCache = context.imageLoader.memoryCache
+        memoryCacheKey?.let { memoryCache?.remove(it) }
+
+        val diskCacheKey = result.diskCacheKey
+        val diskCache = context.imageLoader.diskCache
+        diskCacheKey?.let { diskCache?.remove(it) }
+    }
+
+    return DisposableWrapper(
+        load(url) {
+            user?.let {
+                addHeader(
+                    "Authorization",
+                    ApiUtils.getCredentials(user.username, user.token)
+                )
+            }
+            transformations(CircleCropTransformation())
+            placeholder(R.drawable.account_circle_96dp)
+            listener(onError = { _, result ->
+                Log.w(TAG, "Can't load avatar with URL: $url", result.throwable)
+            })
+        }
+    )
+}
+
+@Deprecated("Use function loadAvatar", level = DeprecationLevel.WARNING)
```

Comment:
only 1 call of this method. can be deleted and use loadAvatar instead?

Reply:
Would leave it in now. @AlvaroBrey  and I will look together next week how we get it all to android-commons.

- https://github.com/nextcloud/talk-android/issues/2559